lib/contentTypeParser.js

```diff
@@ -161,14 +161,28 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
       done
     )
   } else {
-    const result = parser.fn(request, request[kRequestPayloadStream], done)
+    let result
+    try {
+      result = parser.fn(request, request[kRequestPayloadStream], done)
+    } catch (err) {
+      return done(err)
+    }
 
-    if (result && typeof result.then === 'function') {
-      result.then(body => done(null, body), done)
+    if (result) {
+      if (typeof result.then === 'function') {
+        return result.then(body => done(null, body), done)
+      }
+      return done(null, result)
     }
+
+    done(new Error('Cannot call "addContentTypeParser" when fastify instance is already started!'))
```

Comment:
Why?

Reply:
Sorry, I copy pasted the error mechanism from other place on that class, and forgot to update the Error object. Fixed, and thank you for remind me this :-)